dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/update/UpdateRun.java

```diff
@@ -0,0 +1,207 @@
+
+    @CommandLine.Parameters(description = "Update version", arity = "1")
+    private String version;
+
+    @CommandLine.Option(names = { "--openRewriteVersion" },
+                        defaultValue = "6.0.4")
+    String openRewriteVersion;
+
+    @CommandLine.Option(names = { "--camelArtifact" },
+                        defaultValue = "camel-upgrade-recipes")
+    String camelArtifactCoordinates;
+
+    @CommandLine.Option(names = { "--camelSpringBootArtifact" },
+                        defaultValue = "camel-spring-boot-upgrade-recipes")
+    String camelSpringBootArtifactCoordinates;
+
+    @CommandLine.Option(names = { "--camelArtifactVersion" },
+                        defaultValue = "4.8.0")
+    String camelArtifactVersion;
+
+    @CommandLine.Option(names = { "--debug" },
+                        defaultValue = "false")
+    boolean debug;
+
+    @CommandLine.Option(names = { "--quarkusMavenPluginVersion" },
+                        defaultValue = RuntimeType.QUARKUS_VERSION)
+    String quarkusMavenPluginVersion;
+
+    @CommandLine.Option(names = { "--quarkusMavenPluginGroupId" },
+                        defaultValue = "io.quarkus")
+    String quarkusMavenPluginGroupId;
+
+    @CommandLine.Option(names = { "--dryRun" },
+                        defaultValue = "false")
+    boolean dryRun;
+
+    @CommandLine.Option(names = { "--runtime" },
+                        completionCandidates = RuntimeCompletionCandidates.class,
+                        defaultValue = "camel-main",
+                        converter = RuntimeTypeConverter.class,
+                        description = "Runtime (${COMPLETION-CANDIDATES})")
+    RuntimeType runtime = RuntimeType.main;
+
+    @CommandLine.Option(names = { "--repos" },
+                        description = "Additional maven repositories for download on-demand (Use commas to separate multiple repositories)")
+    String repos;
+
+    @CommandLine.Option(names = { "--upgradeTimeout" },
+                        description = "Time to wait, in seconds, before shutting down the upgrade process",
+                        defaultValue = "240")
+    int upgradeTimeout;
+
+    public UpdateRun(CamelJBangMain main) {
+        super(main);
+    }
+
+    /**
+     * Executes the update command for a Camel project. This method determines the appropriate Maven command based on
+     * the runtime type (Camel Main, Spring Boot, or Quarkus) and executes it to update the project to the specified
+     * version.
+     *
+     * @return           the exit code of the update process (0 for success, -1 for failure)
+     * @throws Exception if an error occurs during the update process
+     */
+    @Override
+    public Integer doCall() throws Exception {
+        // Check if the current directory contains a Maven project (i.e., a pom.xml file)
+        if (Files.list(Path.of("."))
+                .filter(f -> f.getFileName().toString().equals("pom.xml"))
+                .count()
+            == 0) {
```

Comment:
maybe you can refactor as Path.of(".")).noneMatch(f -> f.getFileName().toString().equals("pom.xml"))